src/string/src/lib.rs

```diff
@@ -110,8 +117,33 @@ pub unsafe extern "C" fn strcpy(s1: *mut c_char, s2: *const c_char) -> *mut c_ch
 }
 
 #[no_mangle]
-pub extern "C" fn strcspn(s1: *const c_char, s2: *const c_char) -> c_ulong {
-    unimplemented!();
+pub unsafe extern "C" fn strcspn(s1: *const c_char, s2: *const c_char) -> c_ulong {
+    use core::mem;
+
+    let s1 = s1 as *const u8;
+    let s2 = s2 as *const u8;
+
+    // The below logic is effectively ripped from the musl implementation
+
+    let mut byteset = [0u8; 32 / mem::size_of::<usize>()];
+
+    let mut i = 0;
+    while *s2.offset(i) != 0 {
+        byteset[(*s2.offset(i) as usize) / (8 * mem::size_of::<usize>())] |=
```

Comment:
I suggest you to use const variables so as to make this kind of line more readable.

Reply:
You mean something like `const usize_x8 = 8 * mem::size_of::<usize>();`?

Reply:
I guess or something like `const  SIZE_U8  = 8 * mem::size_of::<usize>();` at the global scope.

Reply:
Adding a `const` in any scope is making the postbuild not work.

Reply:
I could just declare it as a variable at the start of the relevant functions (it's only going to be used in two functions)

Reply:
I'm not sure then, I think we should wait on the input of maintainers, maybe they have specific guidelines.

Reply:
It looks ok to me